framework/src/org/checkerframework/common/value/util/Range.java

```diff
@@ -20,6 +20,9 @@
     /** The upper bound of the interval, inclusive. */
     public final long to;
 
+    /** Should ranges take overflow into account or ignore it? */
```

Comment:
I would add a comment calling out at least this basic trade-off.
Using the static mutable field doesn't allow you to use the Range class in different contexts, where one might care about overflow but the other doesn't. That would make composition of different checkers impossible.
I agree that passing around the option isn't nice either.